### PR TITLE
(refactor) Search query format

### PR DESCRIPTION
**Pull request by boi123212321**

### CODE REVIEW

File: src/search/actor.ts [hunk 1]
```diff
@@ -189,8 +209,8 @@ export async function searchActors(
 
   return index.search({
     query: options.query,
-    skip: options.skip || options.page * 24,
-    take: options.take || options.take || PAGE_SIZE,
+    skip: options.skip || (options.page || 0) * 24,
```
Comment (leadwolf) [inline src/search/actor.ts:212]: when there's no skip, wouldn't it be `skip: options.skip || (options.page || 0) * (options.take || PAGE_SIZE)` ?

File: src/search/actor.ts [hunk 2]
```diff
@@ -119,12 +118,27 @@ export async function buildActorIndex(): Promise<Gianna.Index<IActorSearchDoc>>
...
+}
+
 export async function searchActors(
-  query: string,
+  options: Partial<IActorSearchQuery>,
   shuffleSeed = "default",
   transformFilter?: (tree: Gianna.IFilterTreeGrouping) => void
 ): Promise<Gianna.ISearchResults> {
-  const options = extractQueryOptions(query);
+  // const options = extractQueryOptions(query);
```
Comment (leadwolf) [inline src/search/actor.ts:141]: useless comment?

File: src/search/image.ts [hunk 3]
```diff
@@ -203,8 +224,8 @@ export async function searchImages(
 
   return index.search({
     query: options.query,
-    skip: options.skip || options.page * 24,
-    take: options.take || options.take || PAGE_SIZE,
+    skip: options.skip || (options.page || 0) * 24,
```
Comment (leadwolf) [inline src/search/image.ts:227]: same thing about the skip calculation

File: src/search/movie.ts [hunk 4]
```diff
@@ -168,8 +190,8 @@ export async function searchMovies(
 
   return index.search({
     query: options.query,
-    skip: options.skip || options.page * 24,
-    take: options.take || options.take || PAGE_SIZE,
+    skip: options.skip || (options.page || 0) * 24,
```
Comment (leadwolf) [inline src/search/movie.ts:193]: same thing about skip calculation

File: src/search/scene.ts [hunk 5]
```diff
@@ -163,8 +185,8 @@ export async function searchScenes(
 
   return index.search({
     query: options.query,
-    skip: options.skip || options.page * 24,
-    take: options.take || options.take || PAGE_SIZE,
+    skip: options.skip || (options.page || 0) * 24,
```
Comment (leadwolf) [inline src/search/scene.ts:188]: same thing about skip calculation

File: src/search/studio.ts [hunk 6]
```diff
@@ -140,8 +158,8 @@ export async function searchStudios(
 
   return index.search({
     query: options.query,
-    skip: options.skip || options.page * 24,
-    take: options.take || options.take || PAGE_SIZE,
+    skip: options.skip || (options.page || 0) * 24,
```
Comment (leadwolf) [inline src/search/studio.ts:161]: same thing about skip calculation

**Review by leadwolf — CHANGES_REQUESTED** [review]

File: app/src/views/Studios.vue [hunk 7]
```diff
@@ -355,16 +353,16 @@ export default class StudioList extends mixins(DrawerMixin) {
...
       console.error(error);
     }
   }
 
   studioLabels(studio: any) {
-    return studio.labels.map(l => l.name).sort();
+    return studio.labels.map((l) => l.name).sort();
   }
 
   @Watch("ratingFilter", {})
```
[the remark this answers is not in the document]
Comment (boi123212321) [reply app/src/views/Studios.vue:368]: Yeah that should go... copy paste mistake

**Comment by boi123212321** [comment]
TODO markers search

**Review by leadwolf — APPROVED** [review]

**Pull request merged**